modules/web/js/ballerina/env/environment.js

```diff
@@ -179,6 +182,25 @@ class BallerinaEnvironment extends EventChannel {
     }
 
     /**
+     * Initialize operator lattice
+     */
+    initializeOperatorLattice() {
+        return new Promise((resolve, reject) => {
```

Comment:
No need to create a new promise here. You can return the promise received from `getOperatorLatice`.
~~~
return getOperatorLattice()
                 .then((operatorLatticeJson) => {
                     if (operatorLatticeJson) {
                         this._operatorLattice.initFromJson(operatorLatticeJson);
                     } else {
                         log.error('Error while fetching operator lattice');
                     }
                 })
~~~

Reply:
There are no promises returning from getOperatorLattice as we have hardcoded it for the moment. Shall we leave this for now? I'll add a TODO

Reply:
`Promise.resolve` returns a promise here https://github.com/ballerinalang/composer/pull/3335/files#diff-e30e1d6640d67749adfae49b49d69af0R293